test/client.test.ts

```diff
@@ -99,6 +100,22 @@ describe('Tests ReactNativeClient', () => {
       // eslint-disable-next-line deprecation/deprecation
       await expect(RN.YellowBox.ignoreWarnings).toBeCalled();
     });
+    
+    test('use custom transport function', async () => {
+      const myCustomTransportFn = (): Transport => ({
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        send: (request: Envelope) => Promise.resolve(),
+        flush: (timeout?: number) => Promise.resolve(Boolean(timeout))
+      });
+      const client = new ReactNativeClient({
+        ...DEFAULT_OPTIONS,
+        dsn: EXAMPLE_DSN,
+        transport: myCustomTransportFn
+      } as ReactNativeClientOptions);
+
+      await expect(client.getTransport()?.flush(1)).resolves.toBe(true);
+      await expect(client.getTransport()?.send({} as Envelope)).resolves.toBeUndefined();
```

Comment:
If you pulled the Transport send/flush functions to variables you could have an even stronger check of: `expect(client.getTransport()?.flush).toBe(myFlushFunction)`

Reply:
yeah I guess that's another option, shall I change it or it is not critical?

Reply:
🤷 I defer to the Sentry maintainers on that one 😄

Reply:
Yep I'd prefer the stronger check too.

Reply:
done